test/jdk/java/net/httpclient/http2/H2SelectorVTTest.java

```diff
@@ -0,0 +1,222 @@
+            it is configured using an invalid value
+ * @library /test/lib /test/jdk/java/net/httpclient/lib
+ * @build jdk.test.lib.net.SimpleSSLContext
+ *        jdk.httpclient.test.lib.common.HttpServerAdapters
+ * @run junit/othervm
+ *              -Djdk.internal.httpclient.tcp.selector.useVirtualThreads=garbage
+ *              -Djdk.httpclient.HttpClient.log=requests,responses,headers,errors
+ *              H2SelectorVTTest
+ */
+// -Djava.security.debug=all
+class H2SelectorVTTest implements HttpServerAdapters {
+
+    private static SSLContext sslContext;
+    private static HttpTestServer h2Server;
+    private static String requestURI;
+
+    enum UseVTForSelector { ALWAYS, NEVER, DEFAULT }
+    private static final String PROP_NAME = "jdk.internal.httpclient.tcp.selector.useVirtualThreads";
+    private static final UseVTForSelector USE_VT_FOR_SELECTOR;
+    static {
+        String useVtForSelector =
+                System.getProperty(PROP_NAME, "default");
+        USE_VT_FOR_SELECTOR = Stream.of(UseVTForSelector.values())
+                .filter((v) -> v.name().equalsIgnoreCase(useVtForSelector))
+                .findFirst().orElse(UseVTForSelector.DEFAULT);
+    }
+
+    private static boolean isTCPSelectorThreadVirtual() {
+        return switch (USE_VT_FOR_SELECTOR) {
+            case ALWAYS -> true;
+            case NEVER  -> false;
+            default     -> true;
+        };
+    }
+
+    @BeforeAll
+    static void beforeClass() throws Exception {
+        sslContext = new SimpleSSLContext().get();
+        if (sslContext == null) {
+            throw new AssertionError("Unexpected null sslContext");
+        }
+        // create an H3 only server
+        h2Server = HttpTestServer.create(HTTP_2, sslContext);
+        h2Server.addHandler((exchange) -> exchange.sendResponseHeaders(200, 0), "/hello");
+        h2Server.start();
+        System.out.println("Server started at " + h2Server.getAddress());
+        requestURI = "https://" + h2Server.serverAuthority() + "/hello";
+    }
+
+    @AfterAll
+    static void afterClass() throws Exception {
+        if (h2Server != null) {
+            System.out.println("Stopping server " + h2Server.getAddress());
+            h2Server.stop();
+        }
+    }
+
+    /**
+     * Issues various HTTP/2 requests and verifies the responses are received
+     */
+    @Test
+    void testBasicRequests() throws Exception {
+        try (final HttpClient client = HttpClient.newBuilder()
+                .proxy(NO_PROXY)
+                .sslContext(sslContext).build()) {
+            final URI reqURI = new URI(requestURI);
+            final HttpRequest.Builder reqBuilder = HttpRequest.newBuilder(reqURI);
+
+            // GET
+            final HttpRequest req1 = reqBuilder.copy().GET().build();
+            System.out.println("\nIssuing request: " + req1);
+            final HttpResponse<?> resp1 = client.send(req1, BodyHandlers.ofString());
```

Comment:
Nit - this line and the other 3 send() lines could perhaps use `BodyHandlers.discarding()`? If you prefer to use `ofString()`, that's fine too - it wasn't clear to me if this was an oversight.